### PR TITLE
Remote datasets: Add "load_earth_mean_dynamic_topography" to load "CNES Earth Mean Dynamic Topography" dataset

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Add `load_earth_mean_dynamic_topography` to download the "CNES Earth Mean Dynamic Topography" dataset and load into a `xarray.DataArray`:

- Official documentation: https://www.generic-mapping-tools.org/remote-datasets/earth-mdt.html
- Recipe: https://github.com/GenericMappingTools/gmtserver-admin/blob/master/recipes/earth_mdt.recipe

**Adresses** #2431

**Preview**: https://pygmt-dev--3718.org.readthedocs.build/en/3718/api/generated/pygmt.datasets.load_earth_mean_dynamic_topography.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: pygmt/tests/test_datasets_mean_dynamic_topography.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    data = load_earth_mean_dynamic_topography(resolution="01d")
+    assert data.name == "z"
+    assert data.attrs["description"] == "CNES Earth mean dynamic topography"
+    assert data.attrs["units"] == "meters"
+    assert data.attrs["horizontal_datum"] == "WGS84"
+    assert data.shape == (181, 361)
+    assert data.gmt.registration == 0
+    npt.assert_allclose(data.lat, np.arange(-90, 91, 1))
+    npt.assert_allclose(data.lon, np.arange(-180, 181, 1))
+    npt.assert_allclose(data.min(), -1.47, atol=0.01)
```
Comment (seisman) [inline pygmt/tests/test_datasets_mean_dynamic_topography.py:23]: Based on https://github.com/GenericMappingTools/gmtserver-admin/blob/master/recipes/earth_mdt.recipe, the data has a precision of 0.0001 m. So, we should set `atol=0.0001`

File: pygmt/helpers/caching.py [hunk 2]
```diff
@@ -22,6 +22,7 @@ def cache_data():
         "@earth_mag_01d_g",
         "@earth_mag4km_01d_g",
         "@earth_mask_01d_g",
+        "@earth_mdt_01_g",
```
Comment (seisman) [inline pygmt/helpers/caching.py:25]:
```suggestion
        "@earth_mdt_01d_g",
```

File: pygmt/datasets/earth_mean_dynamic_topography.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_mean_dynamic_topography"]
+
+
+def load_earth_mean_dynamic_topography(
+    resolution: Literal["01d", "30m", "20m", "15m", "10m", "07m"] = "01d",
```
Comment (seisman) [inline pygmt/datasets/earth_mean_dynamic_topography.py:18]: I think we need to add a note that `07m` acutally points to a grid with 7.5 arc-minute based on https://github.com/GenericMappingTools/gmtserver-admin/blob/e16841a7260b2c8d075eddc9def69e59569f0cf7/recipes/earth_mdt.recipe#L35.

Actually, we also need to add similar notes to some other datasets like mars_relief, mercury_relief, moon_relief, and pluto_relief.

```
$ grep 'master$' * | awk 'NF==5'
earth_age.recipe:01		m		30		4096	master
earth_dist.recipe:01		m		30		4096	master
earth_edefl.recipe:01		m		30		4096	master
earth_faa.recipe:01		m		30		4096	master
earth_faaerror.recipe:01		m		30		4096	master
earth_gebco.recipe:15		s		10		4096	master
earth_gebcosi.recipe:15		s		10		4096	master
earth_geoid.recipe:01		m		30		4096	master
earth_mag.recipe:02		m		60		4096	master
earth_mag4km.recipe:02		m		60		4096	master
earth_mdt.recipe:7.5		m		0		4096	master
earth_mss.recipe:01		m		30		4096	master
earth_ndefl.recipe:01		m		30		4096	master
earth_relief.recipe:15		s		10		4096	master
earth_synbath.recipe:15		s		10		4096	master
earth_vgg.recipe:01		m		30		4096	master
earth_wdmam.recipe:03		m		90		2048	master
mars_relief.recipe:12.1468873601	s		25.7142857143		4096	master
mercury_relief.recipe:56.25	s		30		4096	master
moon_relief.recipe:14.0625	s		10		4096	master
pluto_relief.recipe:52.0732883317		s		30		4096	master
venus_relief.recipe:01		m		30		4096	master
```

File: pygmt/tests/test_datasets_mean_dynamic_topography.py (hunk 1 above)
Comment (yvonnefroehlich) [reply pygmt/tests/test_datasets_mean_dynamic_topography.py]: Thanks! Updated in commit https://github.com/GenericMappingTools/pygmt/pull/3718/commits/338377c735a24b1b9c6940ba15c73597a528edfe.

File: pygmt/helpers/caching.py (hunk 2 above)
Comment (yvonnefroehlich) [reply pygmt/helpers/caching.py]: Fixed in commits https://github.com/GenericMappingTools/pygmt/pull/3718/commits/f7b660764cbc90d6c3d51194ceeadca8b91fb2ab and https://github.com/GenericMappingTools/pygmt/pull/3717/commits/e7180078ee018020d76a3819f4b360601f1b635c.

File: pygmt/datasets/earth_mean_dynamic_topography.py (hunk 3 above)
Comment (yvonnefroehlich) [reply pygmt/datasets/earth_mean_dynamic_topography.py:18]: Nice catch!
I added a note to the docstring for the `resolution` parameter in commit https://github.com/GenericMappingTools/pygmt/pull/3718/commits/af6cf7aecf7eaa692e970758c48a2249b412dfd1 and submitted PR #3721 for the other four datasets.

File: pygmt/datasets/earth_mean_dynamic_topography.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    Parameters
+    ----------
+    resolution
+        The grid resolution. The suffix ``d`` and ``m`` stand for arc-degrees and
+        arc-minutes. Note that ``"07m"`` refers to a resolution of 7.5 arc-minutes.
+    region
+        The subregion of the grid to load, in the form of a sequence [*xmin*, *xmax*,
+        *ymin*, *ymax*] or an ISO country code. Required for grids with resolutions
+        higher than 5 arc-minutes (i.e., ``"05m"``).
```
Comment (seisman) [inline pygmt/datasets/earth_mean_dynamic_topography.py:59]: No `05m` for this dataset:
```suggestion
        The subregion of the grid to load, in the form of a sequence [*xmin*, *xmax*,
        *ymin*, *ymax*] or an ISO country code.
```

File: pygmt/helpers/caching.py [hunk 5]
```diff
@@ -22,6 +22,7 @@ def cache_data():
         "@earth_mag_01d_g",
         "@earth_mag4km_01d_g",
         "@earth_mask_01d_g",
+        "@earth_mdt_01d_g",
```
Comment (seisman) [inline pygmt/helpers/caching.py:25]: The `07m` data is not tiled.
```suggestion
        "@earth_mdt_01d_g",
        "@earth_mdt_07m_g",
```

File: pygmt/helpers/caching.py [hunk 6]
```diff
@@ -49,6 +50,7 @@ def cache_data():
         "@N00W030.earth_geoid_01m_g.nc",
         "@S30W060.earth_mag_02m_p.nc",
         "@S30W120.earth_mag4km_02m_p.nc",
+        "@N00W030.earth_mdt_07m_g.nc",
```
Comment (seisman) [inline pygmt/helpers/caching.py:53]:
```suggestion
```

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**